### PR TITLE
Store/Retrieve owner info for OneDrive files

### PR DESCRIPTION
**Pull request by meain**
## Description

Store and retrieve owner info for onedrive files.

## Type of change

- [x] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [ ] :computer: CI/Deployment
- [ ] :hamster: Trivial/Minor

## Issue(s)

* https://github.com/alcionai/corso/issues/1366

## Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/connector/onedrive/item.go [hunk 1]
```diff
@@ -67,12 +67,20 @@ func driveItemReader(
 // doesn't have its size value updated as a side effect of creation,
 // and kiota drops any SetSize update.
 func driveItemInfo(di models.DriveItemable, itemSize int64) *details.OneDriveInfo {
+	ed, ok := di.GetCreatedBy().GetUser().GetAdditionalData()["email"]
+
+	email := ""
+	if ok {
+		email = *ed.(*string)
+	}
+
```
Comment (dadams39) [inline src/internal/connector/onedrive/item.go:76]: I believe this is incorrect. Perhaps:
```go
	user  := di.GetCreatedByUser()
	if user != nil{
             
                //  Need to verify that principal name is not empty
		ptr := user.GetUserPrincipalName()
                if ptr != nil{
                          email = *ptr
                          }
	         }
```

File: src/pkg/backup/details/details.go [hunk 2]
```diff
@@ -367,21 +367,22 @@ type OneDriveInfo struct {
...
+	Owner      string    `json:"owner,omitempty"`
 	Created    time.Time `json:"created,omitempty"`
 	Modified   time.Time `json:"modified,omitempty"`
 }
 
 // Headers returns the human-readable names of properties in a OneDriveInfo
 // for printing out to a terminal in a columnar display.
 func (i OneDriveInfo) Headers() []string {
-	return []string{"ItemName", "ParentPath", "Size", "Created", "Modified"}
+	return []string{"ItemName", "ParentPath", "Size", "Owner", "Created", "Modified"}
```
Comment (ryanfkeepers) [inline src/pkg/backup/details/details.go:378]: Does this change need to be reflected in any docs?

Comment (vkamra) [reply src/pkg/backup/details/details.go:378]: I think we're good - I don't see any reference to details anywhere.

**Review by vkamra — APPROVED** [review]
@meain - should be good to merge after the changes @dadams39 suggested are made. Approving so that we can move forward after that.

File: src/pkg/backup/details/details.go (hunk 2 above)
Comment (gmatev) [reply src/pkg/backup/details/details.go:378]: The doc examples are for exchange so this is ok

File: src/internal/connector/onedrive/item.go (hunk 1 above)
Comment (meain) [reply src/internal/connector/onedrive/item.go:76]: Let me look into it more, but `GetCreateByUser` was always `nil` when I tested it.

Comment (meain) [reply src/internal/connector/onedrive/item.go:76]: `GetCreateByUser` seems to be always `nil` but we have data in `GetCreatedBy`. I have made this change locally, but thought I would check once  again with you before pushing it upstream.
/cc @vkamra

Comment (vkamra) [reply src/internal/connector/onedrive/item.go:76]: Let's move forward with this. Why do we need to use `GetAdditonalData` though? What does `GetUser()` return?

Comment (meain) [reply src/internal/connector/onedrive/item.go:76]: `GetUser` only has `DisplayName`.

**Pull request merged**